### PR TITLE
feat(rome_js_parser): Implement option-passing mechanism for parser test #4631

### PR DESCRIPTION
**Pull request by denbezrukov**
## Summary

Close https://github.com/rome/tools/issues/4631

Example: `test test_type test_name {parse_class_parameter_decorators: true/false }`

We need to parse JS parser options from comments and create a corresponding options file.

## Test Plan

`cargo test`

### CODE REVIEW

File: crates/rome_js_parser/Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,8 @@ version              = "0.0.2"
 bitflags              = { workspace = true }
 cfg-if                = "1.0.0"
 drop_bomb             = "0.1.5"
+serde            			= { version = "1", features = ["derive"] }
+serde_json       			= "1"
```
Comment (ematipico) [inline crates/rome_js_parser/Cargo.toml:18]: We can use `workspace = true` for `serde` now

File: crates/rome_js_parser/test_data/inline/err/ts_definite_variable_with_initializer.rast [hunk 2]
```diff
@@ -55,28 +55,32 @@ JsModule {
...
               1: JS_STRING_LITERAL_EXPRESSION@17..23
                 0: JS_STRING_LITERAL@17..23 "\"test\"" [] []
       1: SEMICOLON@23..24 ";" [] []
   3: EOF@24..25 "" [Newline("\n")] []
 --
-ts_definite_variable_with_initializer.js:1:6 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+ts_definite_variable_with_initializer.ts:1:16 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
 
-  × definite assertion assignments are a TypeScript only feature. Convert your file to a TypeScript file or remove the syntax.
+  × Declarations with initializers cannot also have definite assignment assertions.
```
Comment (ematipico) [inline crates/rome_js_parser/test_data/inline/err/ts_definite_variable_with_initializer.rast:73]: Is this change intended?

**Review by ematipico — APPROVED** [review]

File: crates/rome_js_parser/test_data/inline/err/ts_definite_variable_with_initializer.rast (hunk 2 above)
Comment (denbezrukov) [reply crates/rome_js_parser/test_data/inline/err/ts_definite_variable_with_initializer.rast:73]: Yes, I've noticed that it was a mistake.

https://github.com/rome/tools/blob/f65b0d99556daa1f18ddfdea895500a91b09320c/crates/rome_js_parser/src/syntax/stmt.rs#L1294-L1303
